crates/core/tedge_api/src/entity_store.rs

```diff
@@ -179,7 +180,7 @@ impl EntityStore {
             external_id: entity_id.clone(),
             r#type: main_device.r#type,
             parent: None,
-            other: main_device.other,
+            twin_data: main_device.other,
```

Comment:
So `twin_data` stores properties received *either* from registration messages *or* from `twin` message fragments.  Okay.

But, the method `EntityStore::update()` incorrectly overwrite the whole `twin_data` with the `other` field of the  registration message, while `EntityStore::update_twin_data()` correctly updates only the fragment provided by the twin message.

Also, are the data read from a registration message published as twin data to c8y?

Reply:
This overwrite happens only when the entity is registered. You could even say that, that's when the `twin_data` is initialized for the first time. Further updates to `twin/` channel just updates this initial state.

> Also, are the data read from a registration message published as twin data to c8y?

No, we don't want 2 copies of the same data to exist in two different retained messages, leading to unnecessary consistency issues. That's the reason why we're even rejecting updates to `name` and `type` fields via the `twin/` channel.

Reply:
Aah, now I get the issue that you wanted to highlight. Yes, a second registration message for an already registered entity would overwrite the whole twin data with this code. I hadn't considered that. Thought of only the initial registration path. Will fix that.

Reply:
Okay. I would have prefer two distinct `EntityMetadata` fields as the life-cycle and usage of these two json object are so different.

Reply:
I agree, properties related to thin-edge and ones related to C8y are all together and it's hard to know which ones are which. Some properties inside `twin_data` like `name` and `type` need to be treated differently from the other fields, and it's easy to make a mistake like that. We need to think about more clearly separating them.